postgrid/__init__.py

```diff
@@ -145,32 +141,72 @@ def flatten(key, value):
     try:
         res.raise_for_status()
     except requests.HTTPError as e:
-        if base == pm_base:
-            raise PMError(
-                status_code=res.status_code,
-                object=value["object"],
-                type=value["error"]["type"],
-                message=value["error"]["message"],
-            )
-        else:
-            raise NotImplementedError()
+        raise PMError(
+            status_code=res.status_code,
+            object=value["object"],
+            type=value["error"]["type"],
+            message=value["error"]["message"],
+        )
+
+    return _pm_convert_json_value(value)
+
+
+def _av_request(endpoint, intl=False, method="GET", data=None, params=None, json=None):
+    request_endpoint = (intl_av_base if intl else av_base) + endpoint
+
+    data = _map_keys_recursive(data, _snake_to_camel)
+    params = _map_keys_recursive(params, _snake_to_camel)
+    json = _map_keys_recursive(json, _snake_to_camel)
+
+    data_to_pass = []
+    params_to_pass = []
+    files = None
+
+    if data:
+        for key, value in data.items():
+            flatten(files, data_to_pass, key, value)
+    if params:
+        for key, value in params.items():
+            flatten(files, params_to_pass, key, value)
 
-    if base == pm_base:
-        return _pm_convert_json_value(value)
+    headers = {"x-api-key": av_key}
+
+    if method == "GET":
+        res = requests.get(request_endpoint, params=params_to_pass, headers=headers)
+    elif method == "POST":
+        res = requests.post(
+            request_endpoint,
+            json=json,
+            data=data_to_pass,
+            params=params_to_pass,
+            headers=headers,
+        )
+    elif method == "DELETE":
+        res = requests.delete(request_endpoint, params=params_to_pass, headers=headers)
     else:
         raise NotImplementedError()
 
+    value = res.json()
+
+    try:
+        res.raise_for_status()
+    except requests.HTTPError as e:
+        raise AVError(status_code=res.status_code, message=value["message"])
+    if intl:
+        endpoint = "intl/" + endpoint
+    return _av_convert_json_value(endpoint, value)
```

Comment:
Nit: Avoid having to carry along special state where it isn't necessary so just call
`_av_convert_json_value(f"intl/{endpoint}" if intl else endpoint, value)`

Reply:
Noted, fixed.